compatibility/compatibility-tests/src/test/java/org/projectnessie/tools/compatibility/tests/ITUpgradePath.java

```diff
@@ -188,18 +198,130 @@ private void expectedRefLogEntry(String op) {
   }
 
   @Test
-  @Order(5)
+  @Order(105)
   @VersionCondition(minVersion = "0.18.0")
   void refLog() throws Exception {
     List<Tuple> allExpected =
         expectedRefLog.entrySet().stream()
             .flatMap(e -> e.getValue().stream().map(op -> tuple(e.getKey(), op)))
             .collect(Collectors.toList());
 
-    RefLogResponse refLog = api.getRefLog().get();
-    ArrayList<RefLogResponseEntry> logEntries = new ArrayList<>(refLog.getLogEntries());
+    ArrayList<RefLogResponseEntry> logEntries = new ArrayList<>();
+    StreamingUtil.getReflogStream(api, b -> b, OptionalInt.empty()).forEach(logEntries::add);
     Collections.reverse(logEntries);
-    assertThat(logEntries.stream().map(e -> tuple(e.getRefName(), e.getOperation())))
+    assertThat(
+            logEntries.stream()
+                .filter(e -> !keysUpgradeBranch.getName().equals(e.getRefName()))
+                .map(e -> tuple(e.getRefName(), e.getOperation())))
         .containsExactlyElementsOf(allExpected);
   }
+
+  // //////////////////////////////////////////////////////////////////////////////////////////
+  // Keys + values
+  // //////////////////////////////////////////////////////////////////////////////////////////
+
+  private static Branch keysUpgradeBranch;
+  private static final Map<String, Map<ContentKey, IcebergTable>> keysUpgradeAtHash =
+      new LinkedHashMap<>();
+  private static int keysUpgradeSequence;
+  // exceed both number-of-parents per commit-log-entry and per global-log-entry
+  private static final int keysUpgradeCommitsPerVersion = 65;
```

Comment:
should this rather refer to whatever the default is and add 15 to it? I fear that if the default ever gets bumped, then this won't be testing what's expected anymore

Reply:
done